Pcap++/src/PcapFilter.cpp

```diff
@@ -28,15 +28,32 @@ bool GeneralFilter::matchPacketWithFilter(RawPacket* rawPacket)
 	return m_BpfWrapper.matchPacketWithFilter(rawPacket);
 }
 
-BpfFilterWrapper::BpfFilterWrapper()
+namespace internal
 {
-	m_Program = nullptr;
-	m_LinkType = LINKTYPE_ETHERNET;
+	void BpfProgramDeleter::operator()(bpf_program* ptr) const
+	{
+		pcap_freecode(ptr);
+		delete ptr;
+	}
+
+	/**
+	 * @class PcapTDeleter
+	 * A deleter that cleans up a pcap_t structure by calling pcap_close.
+	 */
+	struct PcapTDeleter
+	{
+		void operator()(pcap_t* ptr) const { pcap_close(ptr); }
+	};
```

Comment:
This is nice! I think we can use it in more places in the code, especially in `PcapLiveDevice` and `PcapFileDevice.h`. But for now we can leave this code here